site/content/news/rdf4j-350.md

```diff
@@ -0,0 +1,52 @@
+---
+title: "RDF4J 3.5.0 released"
+date: 2020-12-09T00:00:00+10:00
+layout: "single"
+categories: ["news"]
+---
+We are pleased to announce that RDF4J 3.5.0 is now available. This is a minor release with about 29 improvements and bug fixes.
+
+For full details, have a look at the [release notes](/release-notes/3.5.0). Below, we briefly point out some of the bigger new things.
+
+<!--more-->
+### Easier creation of IRIs and Literals
+
+We've introduced new static factory methods to make your Java code shorter and easier to read. For example, instead of:
+
+```java
+ValueFactory valueFactory = SimpleValueFactory.getInstance();
+
+IRI uncleBob = valueFactory.createIRI("http://example.org/bob");
+IRI lastName = valueFactory.createIRI("http://example.org/name");
+Literal example = valueFactory.createLiteral("Example");
+
+model.add(uncleBob, lastName, example);
+```
+
+you can now use {{< javadoc "Values" "model/util/Values.html" >}} to do this:
+
+```java
+import static org.eclipse.rdf4j.model.util.Values.*;
+
+model.add(iri("http://example.org/bob"), iri("http://example.org/name"), literal("Example"));
+```
+
+See the [RDF Model documentation](/documentation/programming/model) for more details.
+
+### Custom character encoding 
+
+All {{< javadoc "RDFWriter" "rio/RDFWriter.html" >}} and {{< javadoc "QueryResultWriter" "query/resultio/QueryResultWriter.html" >}} implementations for character-based syntax formats now allow the use of a custom writer, with a user-defined custom encoding. Useful if you must produce a CSV file with Windows-encoding instead of the default UTF-8!
+
+### Creation of Literals with java.time values
+
+The new (well, new'ish) `java.time` API is a replacement for the older and clunkier `java.util.Date` and `java.util.Calendar` classes. The RDF4J {{< javadoc "ValueFactory" "model/ValueFactory.html" >}} (as well as the new {{< javadoc "Values" "model/util/Values.html" >}} static factory methods) now supports creation of `Literal` objects directly from `java.time` values, automatically applying the correct XML Schema datatype.
+
+### Peformance improvements
+
+We've made some major improvements in performance of SPARQL querying. Allowing direct result passthrough when querying a remote SPARQL endpoint can result in a reduction of result processing time up to 80-100%. Another area performance has been drastically improved is in the algorithms for ordering and comparing datatyped literals, which makes any SPARQL queries that involve `ORDER BY` clauses significantly quicker.
```

Comment:
Just a note on the order by. By my measurements we see 6x performance improvement when sorting literals, but only ~10% of the time spent on "order by" is spent sorting....the rest is spent creating binding sets. So even though sorting is considerably faster, it doesn't translate to much for SPARQL performance.

Reply:
Good point, I'll adjust a text a little to avoid giving false impressions.